### PR TITLE
Feat/encryption

### PR DESCRIPTION
**Pull request by oed**
Adds support for encryption and decryption using JWEs.
Adds documentation generated with typedoc

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -15,11 +21,34 @@ export async function encodePayload(payload: Record<string, any>): Promise<Encod
...
+export function decodeBase64(s: string): Uint8Array {
+  return u8a.fromString(s, B64)
+}
+
+export function decodeBase58(s: string): Uint8Array {
+  return u8a.fromString(s, B58)
 }
 
-export interface JWSSignature {
+interface JWSSignature {
```
Comment (PaulLeCam) [inline src/utils.ts:51]: Can you make sure it's exported please? I'm using this interface in `idx-tools`

File: src/index.ts [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
+  encodeCleartext,
+  decodeCleartext,
+  encodeBase64,
+  decodeBase64,
+  decodeBase58,
+} from './utils'
 
 export type { DIDDocument } from 'did-resolver'
 
-export type { DagJWS, JWSSignature } from './utils'
```
Comment (PaulLeCam) [inline src/index.ts:8]: Can you make sure it's exported please? I'm using this interface in `idx-tools`

**Review by PaulLeCam — APPROVED** [review]
Nice!
I just noticed the removal of a type export that should be kept if possible please?

**Comment by oed** [comment]
@PaulLeCam Exports added back.

**Comment by PaulLeCam** [comment]
Thanks!

**Review by zachferland — APPROVED** [review]
🎉 👍 

the worst diff, the docs could just be pushed to s3 in the future right?

**Comment by oed** [comment]
Good point @zachferland There is probably even some github action to do that. 
cc @simonovic86

**Pull request merged**